File: TakeFive/takeFive/src/main/java/com/teuskim/takefive/common/CellLocate.java
package com.teuskim.takefive.common;

public class CellLocate {
	private int row, col;
	private int point;
	
	public CellLocate() {}
	
	public CellLocate(int row, int col) {
		this.row = row;
		this.col = col;
	}
	
	public CellLocate(int row, int col, int point) {
		this(row, col);
		this.point = point;
	}

	public int getRow() {
		return row;
	}

	public void setRow(int row) {
		this.row = row;
	}

	public int getCol() {
		return col;
	}

	public void setCol(int col) {
		this.col = col;
	}

	public int getPoint() {
		return point;
	}

	public void setPoint(int point) {
		this.point = point;
	}

	@Override
	public boolean equals(Object o) {
		CellLocate other = (CellLocate) o;
		return (this.row==other.row && this.col==other.col);
	}

	@Override
	public int hashCode() {
		return row+(col*10000);
	}

	@Override
	public String toString() {
		return "row:"+row+", col:"+col;
	}
	
}

File: TakeFive_kotlin/takeFive/src/main/java/com/teuskim/takefive/common/CellLocate.kt
package com.teuskim.takefive.common

class CellLocate {
    var row: Int = 0
    var col: Int = 0
    var point: Int = 0

    constructor() {}

    constructor(row: Int, col: Int) {
        this.row = row
        this.col = col
    }

    constructor(row: Int, col: Int, point: Int) : this(row, col) {
        this.point = point
    }

    override fun equals(o: Any?): Boolean {
        val other = o as CellLocate?
        return this.row == other!!.row && this.col == other.col
    }

    override fun hashCode(): Int {
        return row + col * 10000
    }

    override fun toString(): String {
        return "row:$row, col:$col"
    }

}